build.rs

```diff
@@ -0,0 +1,249 @@
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::fs::File;
+use std::io::{BufReader, BufWriter, Write};
+use std::path::PathBuf;
+use std::{env, io};
+
+use serde::Deserialize;
+
+enum Table {
+    Complete([char; 128]),
+    Incomplete([Option<char>; 128]),
+}
+
+struct CodeTables {
+    created: String,
+    tables: Vec<(u16, Table)>,
+}
```

Comment:
I called it `CodeTables`, because the file is called `code_tables.json`. I also wanted to avoid a long name.

I chose `Vec`, because I want the tables to be sorted and stable in the generated output.

Reply:
@mkroening Autocompletion by modern editors reduces a pain of typing long names a lot, though.

> because I want the tables to be sorted and stable in the generated output.

Why not `BTreeMap`?

Reply:
@mkroening I'm glad to hear your opinion. You don't have to modify.

Reply:
They should scale the same, and `BTreeMap` makes sure that the collection is kept sorted. `Vec` provides better cache locality, since its items are stored in a row. So it's about nice access patterns when iterating. For the `BTreeMap`, we'd jump around in memory a lot.

All of this is super irrelevant for our use case, since this is not at all performance sensitive, and benchmarks might prove me wrong in this case. So my choice was rather arbitrary and mainly out of habit. :)